front/app/components/admin/ActionForm/FlowVisualization/Blocks/VerificationBlock.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+
+import {
+  Box,
+  colors,
+  stylingConsts,
+  Icon,
+} from '@citizenlab/cl2-component-library';
+import styled from 'styled-components';
+
+import useVerificationMethod from 'api/verification_methods/useVerificationMethod';
+
+import useLocalize from 'hooks/useLocalize';
+
+import { useIntl } from 'utils/cl-intl';
+
+import messages from './messages';
+import { getReturnedFieldsPreview } from './utils';
+import VerificationModal from './VerificationModal';
+
+const StyledBox = styled(Box)`
+  &:hover {
+    background-color: ${colors.teal100};
+  }
+`;
+
+interface Props {
+  number: number;
+}
+
+const VerificationBlock = ({ number }: Props) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const { formatMessage } = useIntl();
+  const localize = useLocalize();
+  const { data: verificationMethod } = useVerificationMethod();
+
+  const verificationMethodMetadata =
+    verificationMethod?.data.attributes.action_metadata;
+
+  if (!verificationMethodMetadata) return null;
+
+  const verifiedFieldsMessage = formatMessage(messages.verifiedFields);
+
+  const returnedFieldsPreview = getReturnedFieldsPreview(
+    verificationMethodMetadata,
+    localize
+  );
+
+  return (
```

Comment:
This is this thing (number 3):

![Screenshot 2024-08-27 at 15 55 49](https://github.com/user-attachments/assets/de591104-2060-4a8b-b2f5-666b6a666b85)